### DISCUSSION TITLE
"Config needs migrating" but shows no steps needing migration?

### DISCUSSION THREAD

**Discussion by jamietanna** [renovatebot/renovate#19960] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.97.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

The config shown has changes that can be performed.

### What you tried so far.

Shows:

```json
 WARN: Config needs migrating
       "originalConfig": {
         "username": "...",
         "gitAuthor": "...",
         "requireConfig": "optional",
         "onboarding": false,
         "platform": "github",
         "includeForks": true,
         "extends": ["github>org/presets"],
         "hostRules": [{"hostType": "docker"}]
       },
       "migratedConfig": {
         "username": "...",
         "gitAuthor": "...",
         "requireConfig": "optional",
         "onboarding": false,
         "platform": "github",
         "includeForks": true,
         "extends": ["github>org/presets"],
         "hostRules": [{"hostType": "docker"}]
       }
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
This looks like global config? How/when exactly is this being validated?

**Reply by jamietanna** [reply]
Yep, this is global config - I see it when I run i.e. `renovate --dry-run jamietanna/dotfiles-arch` from the directory that my `config.js` containing global config is found

**Reply by viceice** [reply]
seeing this too for global config bit didn't had time yet to analyze the cause. if I compare both JSON objects then they ate equal. 🤷‍♂️

**Reply by rarkins** [reply]
I'm not seeing this in mine locally

**Reply by rarkins** [reply]
https://github.com/renovatebot/renovate/blob/35ff7e12109c10f377c82e13ad02c9f15cf3430f/lib/config/migration.ts#L146

**Reply by viceice** [reply]
ok, this time it was my bad. wrong dry-run config.

**Comment by jamietanna** [comment, marked as answer]
I've worked it out that it's that _inside_ the `extends` there's a `includeForks: true` that needs migrating.

Would be good if we can try and highlight that it may be an `extends` that's at fault in this case, as it could be confusing - maybe in the case that `originalConfig == migratedConfig`?

**Reply by rarkins** [reply]
I think we should differentiate between "YOUR config needs migrating" (which is actionable) versus "An extends you depend on needs migrating" (which may not be actionable)

**Reply by HonkingGoose** [reply]
See issue:

- #21456
